Neural Networks/01.MNIST Classification/image_aug.py

```diff
@@ -0,0 +1,92 @@
+import utils
+import numpy as np
+import matplotlib.pyplot as plt
+from skimage.filters import difference_of_gaussians, unsharp_mask, gaussian
+from skimage.transform import rescale, resize, downscale_local_mean
+import gzip
+import idx2numpy
+import os
+
+# input files and scale input data
+with gzip.open(os.getcwd() + '\\00_MNIST_data\\t10k-images-idx3-ubyte.gz', 'r') as f:
+    test_images = idx2numpy.convert_from_file(f)
+
+scaled_test_images = test_images / 255.
+
+# randomly chose an image from test set then plot image
+random_inx = np.random.choice(scaled_test_images.shape[0])
+# random_inx = 8700 # 9828
+test_image = scaled_test_images[random_inx]
+
+# crop image
+cropped_img = utils.remove_image_margins(test_image)
+
+# apply a Gaussian blur filter
+filtered_cr_img = gaussian(test_image, sigma=1.1)
```

Comment:
Might be useful to make the noise scale with pixel values, or be different for different pixel values. You can pass an array for `sigma`. 

Also you could simply add a `numpy.random` array instead of `skimage`.